external-crates/move/crates/move-compiler/src/command_line/compiler.rs

```diff
@@ -96,21 +114,44 @@ enum PassResult {
     Typing(typing::ast::Program),
     HLIR(hlir::ast::Program),
     CFGIR(cfgir::ast::Program),
-    Compilation(Vec<AnnotatedCompiledUnit>, /* warnings */ Diagnostics),
+    Compilation(
+        Vec<AnnotatedCompiledUnit>,
+        Vec<ModuleIdent>,
+        /* warnings */ Diagnostics,
+    ),
 }
 
+pub type CompiledModuleInfoMap = BTreeMap<ModuleIdent, Arc<CompiledModuleInfo>>;
+
 #[derive(Clone)]
-pub struct FullyCompiledProgram {
-    pub files: MappedFiles,
-    pub parser: parser::ast::Program,
-    pub expansion: expansion::ast::Program,
-    pub naming: naming::ast::Program,
-    pub typing: typing::ast::Program,
-    pub hlir: hlir::ast::Program,
-    pub cfgir: cfgir::ast::Program,
-    pub compiled: Vec<AnnotatedCompiledUnit>,
+pub struct CompiledModuleInfo {
+    pub file_name: Symbol,
+    pub file_content: Arc<str>,
+    /// to compute address conflicts in expansion/translate.rs
+    /// (address map "inlined" into the module so that
+    /// modules are separated)
+    pub named_address_map: NamedAddressMap,
+    /// to create context in expansion/translate.rs
+    pub member_kinds: BTreeMap<Name, ModuleMemberKind>,
+    /// to build context in naming/translate.rs
+    pub resolved_members: BTreeMap<Symbol, ResolvedModuleMember>,
+    /// to extract macros in typing/translate.rs (need function bodies for this)
+    pub macro_infos: Option<(UseFuns, UniqueMap<FunctionName, Function>)>,
+    /// to construct program infos in shared/program_info.rs
+    /// (need both because typing info drops some info present at naming,
+    /// such as abilities, and naming info does not have all the types resolved)
+    pub naming_info: ModuleInfo,
+    pub typing_info: ModuleInfo,
```

Comment:
We really shouldn't duplicate these. They are rather large.

> typing info drops some info present at naming, such as abilities,

This seems surprising to me. Typing information should be strictly greater

> naming info does not have all the types resolved

But that seems fine? How would that be an issue when repopulating this information during naming?

Reply:
Cool. I wasn't sure what the intention was here for these being separately crated for naming and typing. For a bit of additional context when I used `naming` one only, I was getting a failure in one of the linters due to unresolved type and when I was using `typing` only, I was seeing compiler transactional tests failures due to missing abilities. Happy to investigate the discrepancy and add things (back) to `typing` if need be